### PR TITLE
feat(reminders): add verification reminders

### LINKED ISSUE #99: Add support for sending "Reminder to Verify" emails
Related to https://github.com/mozilla/fxa-auth-server/issues/1081

### PR DESCRIPTION
**Pull request by vladikoff**
Fixes #99

### CODE REVIEW

File: lib/verification-reminders.js [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+        // avoids spamming multiple emails
+        return db.deleteReminder(reminder)
+          .then(function () {
+            // fetch the account
+            return db.emailRecord(reminder.email)
+          })
+          .then(function (account) {
+            if (! account.emailVerified) {
+              // if account is not verified then send the reminder
+              mailer.verificationReminderEmail(reminder)
```
Comment (vladikoff) [inline lib/verification-reminders.js]: TODO: code is empty in mysql mode

File: lib/db.js [hunk 2]
```diff
@@ -48,5 +49,48 @@ module.exports = function () {
       )
   }
 
+  DB.prototype.account = function (uid) {
+    log.trace({ op: 'DB.account', uid: uid })
```
Comment (vladikoff) [inline lib/db.js]: Check LOG_LEVEL in Webstorm....

File: lib/verification-reminders.js [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+var config = require('../config')
+var reminderConfig = config.get('verificationReminders')
+
+module.exports = function (mailer, db, options) {
+  options = options || {}
+  var log = options.log || logger
+
+  return {
+    /**
+     *
```
Comment (vbudhram) [inline lib/verification-reminders.js]: I think you might have missed a function description comment here.

File: lib/verification-reminders.js [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+     * @param reminderData
+     * @param {buffer} reminderData.uid - The uid to remind.
+     * @param {string} reminderData.type - The type of a reminder.
+     * @returns {Promise}
+     * @private
+     */
+    _processReminder: function (reminderData) {
+        log.debug('_processReminder', reminderData)
+        // only send email once we safely delete the reminder
```
Comment (vbudhram) [inline lib/verification-reminders.js]: :sunglasses:

File: test/local/reminder_db_tests.js [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+      })
+      .then(
+        function () {
+          return db.fetchReminders({
+            // fetch reminders older than 'reminderTime'
+            reminderTime: 1,
+            reminderTimeOutdated: 5000,
+            type: 'first',
+            limit: 200
+          }).catch(function (err) {
```
Comment (vbudhram) [inline test/local/reminder_db_tests.js]: Any reason to catch error and then re-throw error? Could you put the catch at the bottom of promise chain and then fail test if error?

**Comment by vladikoff** [comment]
Rebased and addressed review feedback

**Pull request merged**